controllers/servers/host_definer/kubernetes_manager/manager.py

```diff
@@ -169,27 +169,21 @@ def _is_host_definition_matches(self, host_definition_info, node_name, secret_na
             host_definition_info.secret_name == secret_name and \
             host_definition_info.secret_namespace == secret_namespace
 
-    def _patch_host_definition(self, host_definition_manifest):
-        host_definition_name = host_definition_manifest[settings.METADATA][settings.NAME]
-        logger.info(messages.PATCHING_HOST_DEFINITION.format(host_definition_name))
-        try:
-            self.host_definitions_api.patch(body=host_definition_manifest, name=host_definition_name,
-                                            content_type='application/merge-patch+json')
-        except ApiException as ex:
-            if ex.status == 404:
-                logger.error(messages.HOST_DEFINITION_DOES_NOT_EXIST.format(host_definition_name))
-            else:
-                logger.error(messages.FAILED_TO_PATCH_HOST_DEFINITION.format(host_definition_name, ex.body))
-
     def _create_host_definition(self, host_definition_manifest):
         try:
             k8s_host_definition = self.host_definitions_api.create(body=host_definition_manifest)
+            self._add_finalizer_to_host_definition(k8s_host_definition.metadata.name)
             return self._generate_host_definition_info(k8s_host_definition)
         except ApiException as ex:
             if ex != 404:
                 logger.error(messages.FAILED_TO_CREATE_HOST_DEFINITION.format(
                     host_definition_manifest[settings.METADATA][settings.NAME], ex.body))
 
+    def _add_finalizer_to_host_definition(self, host_definition_name):
```

Comment:
I think _add_finalizer is enough (as the function gets `host_definition_name`).
same for update

Reply:
done